AppRun.c

```diff
@@ -37,6 +37,12 @@ THE SOFTWARE.
         fprintf(stderr, "Error: " __VA_ARGS__);     \
         exit(1);                                    \
     } while(0);
+#define SET_NEW_ENV(str,len,fmt,...)                \
+    format = fmt;                                   \
+    length = strlen(format) + (len);                \
+    char *str = malloc(length);                     \
```

Comment:
Maybe I should use `calloc()` or calculate the size of the `%s` chars out.

Reply:
I don't see what `calloc` would add to this. You're effectively overwriting the contents anyway. The only problem I'd say `calloc` would be useful is if the resulting string (i.e.,`format`) was empty, because then, the string wouldn't be terminated properly.

Reply:
There's 2 unused bytes for each `%s`, for example:
format = "PYTHONHOME=%s/usr/"
strlen(format) = 18
appdir = "/tmp/.mount_APPDIR"
strlen(appdir) = 18
length = strlen(format) + strlen(appdir) = 36
strlen("PYTHONHOME=/tmp/.mount_APPDIR/usr/") = 34

Or maybe use `memset()` right after `malloc()`?

Reply:
I guess `calloc()` will suffice, it'll provide the functionality needed here (properly terminated strings) and a combination of `memset` and `malloc` won't be faster but more error prone.